src/API/Site/Controllers/Ads/AssetGroupController.php

```diff
@@ -0,0 +1,204 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\AdsAssetGroup;
+use WP_REST_Request as Request;
+use Exception;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AssetGroupController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads
+ */
+class AssetGroupController extends BaseController {
+
+	/**
+	 * The AdsAssetGroup class.
+	 *
+	 * @var AdsAssetGroup $ads_asset_group
+	 */
+	protected $ads_asset_group;
+
+	/**
+	 * AssetGroupController constructor.
+	 *
+	 * @param RESTServer    $rest_server
+	 * @param AdsAssetGroup $ads_asset_group
+	 */
+	public function __construct( RESTServer $rest_server, AdsAssetGroup $ads_asset_group ) {
+		parent::__construct( $rest_server );
+		$this->ads_asset_group = $ads_asset_group;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'ads/campaigns/(?P<id>[\d]+)/assetgroups',
```

Comment:
Some naming/location about this controller. Probably I'm missing concepts. But just to be sure :) 
 
❓  Why this file is not under Controller/Assets as `AssetSuggestionsController`
❓  I would name it under `assets/(?P<id>[\d]+)/groups` 

💅  In any case, instead of `assetgroups` I'd suggest `asset-groups`

Reply:
> ❓ Why this file is not under Controller/Assets as AssetSuggestionsController

I have some doubts about the location too, my thinking was Asset Groups are linked to a campaign, campaign are in the `Ads` directory, and asset groups belongs as well to the "Ads" section, and I was thinking as well about moving the `AssetSuggestionsController ` to the `Ads` directory. WDYT? Or do you think it is better to leave a specific folder for the `Assets`?

> ❓ I would name it under assets/(?P<id>[\d]+)/groups

Not sure about this one, because it seems that groups are under `assets` but the hierarchy is more like this:

Campaign -> Asset Group -> Asset Group Asset (the link between Asset Group and Asset) -> Assets. Also, the terminology in the Ads API is `Asset Group`.

> 💅 In any case, instead of assetgroups I'd suggest asset-groups

👍